File: engine/src/main/java/com/terralite/engine/physics/CollisionPair.java
package com.terralite.engine.physics;

import com.terralite.engine.entity.Entity;

import java.util.Objects;

public record CollisionPair(Entity first, Entity second) {
    public CollisionPair {
        Objects.requireNonNull(first, "first");
        Objects.requireNonNull(second, "second");
        if (first == second || first.id().equals(second.id())) {
            throw new IllegalArgumentException("Collision pair requires two distinct entities");
        }
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/physics/CollisionPair.kt
package com.terralite.engine.physics

import com.terralite.engine.entity.Entity

@JvmRecord
data class CollisionPair(val first: Entity, val second: Entity) {
    init {
        require(first !== second && first.id != second.id) {
            "Collision pair requires two distinct entities"
        }
    }
}
